Screenshoddy/UI/Preferences/Account/AccountPreferencesViewController.swift

```diff
@@ -45,6 +48,13 @@ class AccountPreferencesViewController: NSViewController {
         print("Changed")
     }
     
+    @objc func getBucketsButtonClicked(_ sender: NSButton) {
+        // Fetch s3 buckets that the user-supplied keys have access to.
+        let s3 = S3(accessKeyId: "Your-Access-Key", secretAccessKey: "Your-Secret-Key")
```

Comment:
You can access preferences for `accessKeyId` and `secretAccessKey` with:

- `AppDefaults.s3AccessId`
- `AppDefaults.s3SecretKey`

Reply:
Actually, those preferences won't be updated quite yet to fetch initial buckets (The preferences only get saved on window close). You could grab the direct string values from the fields via `accessIdField.stringValue` and `secretKeyField.stringValue`.